components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -785,32 +744,20 @@ void RewardsServiceImpl::OnLoad(SessionID tab_id, const GURL& url) {
     return;
   }
 
-  if (!ProcessPublisher(url)) {
+  if (IsAutoContributeHandledByContentScript(url)) {
     return;
   }
 
-  auto origin = url.host();
-  std::string baseDomain =
-      GetDomainAndRegistry(url.host(), INCLUDE_PRIVATE_REGISTRIES);
-#if BUILDFLAG(ENABLE_IPFS)
-  if (baseDomain.empty()) {
-    baseDomain = ipfs::GetRegistryDomainFromIPNS(url);
-    if (!baseDomain.empty()) {
-      origin = baseDomain;
-    }
-  }
-#endif
-  if (baseDomain == "")
+  auto publisher_domain = GetPublisherDomainFromURL(url);
+  if (!publisher_domain) {
     return;
-
-  const std::string publisher_url = url.scheme() + "://" + baseDomain + "/";
+  }
 
   ledger::mojom::VisitDataPtr data = ledger::mojom::VisitData::New();
-  data->tld = data->name = baseDomain;
-  data->domain = origin;
+  data->domain = data->tld = data->name = *publisher_domain;
```

Comment:
It seems weird that the domain and TLD fields would both contain the same value. It doesn't seem like we'd actually need to track TLD at all really, although maybe I'm missing a subtlety there.

Reply:
Good call - I removed the `tld` field completely.